updater/lib/dependabot/updater/operations/refresh_version_pull_request.rb

```diff
@@ -0,0 +1,346 @@
+# the existing pull request with a new one for clarity.
+module Dependabot
+  class Updater
+    module Operations
+      class RefreshVersionPullRequest
+        def self.applies_to?(job:)
+          return false if job.security_updates_only?
+          # If we haven't been given metadata about the dependencies present
+          # in the pull request, this strategy cannot act.
+          return false if job.dependencies&.none?
+
+          job.updating_a_pull_request?
+        end
+
+        def initialize(service:, job:, dependency_snapshot:, error_handler:)
+          @service = service
+          @job = job
+          @dependency_snapshot = dependency_snapshot
+          @error_handler = error_handler
+        end
+
+        def perform
+          Dependabot.logger.info("Starting PR update job for #{job.source.repo}")
+          dependency = dependencies.last
+          check_and_update_pull_request(dependencies)
+        rescue StandardError => e
+          error_handler.handle_dependabot_error(error: e, dependency: dependency)
+        end
+
+        private
+
+        attr_reader :job,
+                    :service,
+                    :dependency_snapshot,
+                    :error_handler,
+                    :created_pull_requests
+
+        def dependencies
+          all_deps = dependency_snapshot.dependencies
+
+          # Rebases and security updates have dependencies, version updates don't
+          if job.dependencies
+            # Gradle, Maven and Nuget dependency names can be case-insensitive and
+            # the dependency name in the security advisory often doesn't match what
+            # users have specified in their manifest.
+            #
+            # It's technically possibly to publish case-sensitive npm packages to a
+            # private registry but shouldn't cause problems here as job.dependencies
+            # is set either from an existing PR rebase/recreate or a security
+            # advisory.
+            job_dependencies = job.dependencies.map(&:downcase)
+            return all_deps.select do |dep|
+              job_dependencies.include?(dep.name.downcase)
+            end
+          end
+
+          allowed_deps = all_deps.select { |d| job.allowed_update?(d) }
+          # Return dependencies in a random order, with top-level dependencies
+          # considered first so that dependency runs which time out don't always hit
+          # the same dependencies
+          allowed_deps = allowed_deps.shuffle unless ENV["UPDATER_DETERMINISTIC"]
+
+          if all_deps.any? && allowed_deps.none?
+            Dependabot.logger.info("Found no dependencies to update after filtering allowed " \
+                                   "updates")
+          end
+
+          allowed_deps
+        end
+
+        # rubocop:disable Metrics/AbcSize
+        # rubocop:disable Metrics/PerceivedComplexity
+        #
+        # TODO: Push checks on dependencies into Dependabot::DependencyChange
```

Comment:
I'm copying this over from the Updater where the TODO was originally marked out. I'm not sure it applies in the way I thought it might, I was thinking that as we filtered and set the dependencies, we could do some of the work in that class as a kind of batch compiler, so we'd set @dependencies, then verify!, then generate the change all on that object.

I'm not sure that works anymore as there's a pattern emerging between "single" and "batch" operations I hadn't really thought much about - i.e. security updates, and all kinds of rebases are single, version updates (and groups) are batches - so this might make more sense fitting into that.